develop/tutorials/articles/290-configuration/01-making-apps-configurable/06-custom-system-settings-ui.markdown

```diff
@@ -0,0 +1,229 @@
+    them in a Map whose keys should be the names of the fields of the
+    Configuration interface.
+
+3.  `render`: Render the custom form's fields, using your desired method (for
+    example, JSPs or another template mechanism). The `<form>` tag itself is
+    provided automatically and shouldn't be included in the
+    `ConfigurationFormRenderer`.
+
+Here's a complete `ConfigurationFormRenderer` implementation:
+
+    @Component(immediate = true, service = ConfigurationFormRenderer.class)
+    public class CurrencyConverterConfigurationFormRenderer
+        implements ConfigurationFormRenderer {
+
+        @Override
+        public String getPid() {
+
+            return "com.liferay.currency.converter.web.configuration.CurrencyConverterConfiguration";
+        }
+
+        @Override
+        public void render(HttpServletRequest request, HttpServletResponse response)
+            throws IOException {
+
+            String formHtml = "<input name=\"mysymbols\" />";
+
+            PrintWriter writer = response.getWriter();
+
+            writer.print(formHtml);
+
+        }
+
+        @Override
+        public Map<String, Object> getRequestParameters(
+            HttpServletRequest request) {
+
+            Map<String, Object> params = new HashMap<>();
+
+            String[] mysymbols = ParamUtil.getParameterValues(request, "mysymbols");
+
+            params.put("symbols", mysymbols);
+
+            return params;
+        }
+    }
+
+The above example generates a custom rendering (HTML) for the form in the
+`render()` method and reads the information entered in the custom form in the
+`getRequestParameters()` method.
+
+To see a complete demonstration, including JSP markup, read the dedicated
+tutorial on creating a 
+[configuration form renderer](/develop/tutorials/-/knowledge_base/7-1/configuration-form-renderer).
+
+## Creating a Completely Custom Configuration UI [](id=creating-a-completely-custom-configuration-ui)
+
+In some cases, you want a completely custom UI for your configuration. For
+example: 
+
+- Your application doesn't use Config Admin to provide its configuration. You
+    have a completely different configuration backend, and you'll write a
+    completely independent frontend.
+
+- Your application needs more flexibility in its UI, such as multiple
+    configuration screens.
+
+To accomplish this, write a `ConfigurationScreen` implementation.
+
+At a high level you must
+
+1.  Write a Component that declares itself an implementation of the
+    `ConfigurationScreen` interface.
+
+2.  Implement `Configurationscreen`'s methods.
+
+3.  Create the UI by hand.
+
+Here's an example implementation:
+
+    @Component(immediate = true, service = ConfigurationScreen.class) 
+    public class SampleConfigurationScreen implements ConfigurationScreen {
+
+First declare the class an implementation of `ConfigurationScreen`.
+
+    @Override 
+    public String getCategoryKey() { 
+
+        return "third-party"; 
+        
```

Comment:
Is there a reason these return statements are padded with newlines?  I see in the example code for `07-fully-custom-system-settings-ui.markdown` that they don't have newlines above and below, so it looks inconsistent to me.